Common/Algorithm/Framework/Portfolio/SignalExports/BaseSignalExport.cs

```diff
@@ -0,0 +1,88 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Interfaces;
+using QuantConnect.Logging;
+using QuantConnect.Util;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+
+namespace QuantConnect.Algorithm.Framework.Portfolio.SignalExports
+{
+    /// <summary>
+    /// Base class to send signals to different 3rd party API's
+    /// </summary>
+    public abstract class BaseSignalExport : ISignalExportTarget
+    {
+        /// <summary>
+        /// Lazy initialization of HttpClient to be used to sent signals to different 3rd party API's
+        /// </summary>
+        private Lazy<HttpClient> _lazyClient = new Lazy<HttpClient>();
+
+        /// <summary>
+        /// Property to access a HttpClient
+        /// </summary>
+        protected HttpClient HttpClient => _lazyClient.Value;
+
+        /// <summary>
+        /// Default hashset of allowed Security types for Numerai and CrunchDAO API's
+        /// </summary>
+        protected readonly HashSet<SecurityType> DefaultAllowedSecurityTypes = new()
```

Comment:
Please make DefaultAllowedSecurityTypes overridable and a full security list then override narrower the list on subclasses. Its not ideal only applicable on 2/3 subclasses